evalml/data_checks/highly_null_rows_data_check.py

```diff
@@ -0,0 +1,85 @@
+from evalml.data_checks import (
+    DataCheck,
+    DataCheckAction,
+    DataCheckActionCode,
+    DataCheckMessageCode,
+    DataCheckWarning
+)
+from evalml.utils import _convert_woodwork_types_wrapper, infer_feature_types
+
+
+class HighlyNullRowsDataCheck(DataCheck):
+    """Checks if there are any highly-null rows in the input."""
+
+    def __init__(self, pct_null_threshold=0.5):
```

Comment:
I set default to `0.5` because I think this can be more aggressive on removing individual samples/rows than removing a whole column in `HighlyNullDataCheck`.